### DISCUSSION TITLE
How to use monthly / seasonal climatic_mass_balance as forcing?

### DISCUSSION THREAD

**Discussion by jacksd233** [pism/pism#538] [Q&A]
Hello,
    I have used constant climatic_mass_balance to run PISM. Now I want to try monthly changing climatic_mass_balance (e.g. climatic_mass_balance: [time=12, mx, my]) as forcing but don't know how to do it.

**Comment by talbrecht** [comment]
Hej,
have you checked the manual https://www.pism.io/docs/climate_forcing/time-dependent.html#id8 ?

**Reply by jacksd233** [reply]
yes, I saw the 'Periodic forcing' part. However there is no example of periodic forcing, so I am slightly confused about how to 'turn on' periodic forcing.
For example, If I want to add a monthly changing forcing and run for 100 years, how should I set the time in the forcing file? The manual says 'A PISM forcing file with a periodic forcing has to contain exactly one period'. Does it mean that just setting 12 time for each month will turn on periodic forcing?

**Reply by juliusgarbe** [reply]
Hi @jacksd233, you are right. Seeing your question I just realized that the manual is not entirely complete with regards to how the periodic forcing is actually turned on in PISM. We should definitely add this, thanks for the reminder!

In principle, there are two ways to turn on periodic forcing in PISM: 

**(1) Via a config parameter**
E.g. by setting a flag parameter such as `surface.given.periodic` or `atmosphere.delta_T.periodic` or similar to *yes* (they are *no* by default). To find more similar config parameters that are available, you could for example search for 'periodic' on [this manual page](https://www.pism.io/docs/manual/parameters/index.html).

**(2) Via a PISM option / argument**
For most of these above mentioned flags, there should be a corresponding PISM option / argument. For example, if you are using an atmosphere coupler such as `-atmosphere given`, just add `-atmosphere_given_period 1`. The '1' tells PISM that the period of the forcing file is 1 year. The option `-atmosphere_given_file` should then point to a forcing file that contains exactly one period (in your case, for example, the twelve months), as described [here](https://www.pism.io/docs/climate_forcing/time-dependent.html#periodic-forcing).

Note also the specific requirements for the periodic forcing file that are given in the manual. For example, it's recommended to use the `365_day` calendar, starting at January 1, 1. Accordingly, the `time_bnds` must be set properly, as described in the manual.

The forcing file should look something like this:

```
netcdf forcing {
dimensions:
	time = UNLIMITED ; // (12 currently)
	bnds = 2 ;
	x = 761 ;
	y = 761 ;
	nv4 = 4 ;
variables:
	double time_bnds(time, bnds) ;
	double time(time) ;
		time:axis = "T" ;
		time:calendar = "365_day" ;
		time:long_name = "time" ;
		time:standard_name = "time" ;
		time:units = "days since 01-01-01 00:00:00.0" ;
		time:bounds = "time_bnds" ;
[...]
data:
	time = 15, 44.5, 74, 104.5, 135, 165.5, 196, 227, 257.5, 288, 318.5, 349 ;
	time_bnds =
  0, 32,
  32, 60,
  60, 91,
  91, 121,
  121, 152,
  152, 182,
  182, 213,
  213, 244,
  244, 274,
  274, 305,
  305, 335,
  335, 366 ;
}
```

In this example, the `time` variable contains the mid-points of each month of the 365-days year. The `time_bnds` are always start and end of each month, respectively.

Hope this helps. If you have more questions, let me know!
